contracts/signature_mint/SignatureMint.sol

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+// Interface
+import { ISignatureMint } from "./ISignatureMint.sol";
+
+// Token
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+
+// Protocol control center.
+import { ProtocolControl } from "../ProtocolControl.sol";
+
+// Signature utils
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+// Access Control + security
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+// Royalties
+import "@openzeppelin/contracts/interfaces/IERC2981.sol";
+
+// Meta transactions
+import "@openzeppelin/contracts/metatx/ERC2771Context.sol";
+
+// Utils
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/Multicall.sol";
+
+contract SignatureMint is
```

Comment:
I think we should include 721 in the contract so at least we can expand into 1155 in the future.
SignatureMintERC721 or something along the line